### PR TITLE
[114] Implement actor create resumes API @POST

### PR DESCRIPTION
**Pull request by badnameok**
## What did you do

- Implemented POST resume API

## Demo

```
{
  "name": "test resume v1",
  "resumeUrl": "google.com"
}
```
The above should work provided that you are properly logged into a verified actor account.
I suggest slightly modifying to make a v2 and trying again to see if it properly appends.
```
{
  "name": "test resume v3",
  "resumeUrl": "thisisnotaurl"
}
```
The above should raise a bad request exception due to the url being invalid.

### CODE REVIEW

File: apps/api/src/modules/resume/resume.controller.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  ApiTags,
+  ApiUnauthorizedResponse,
+} from '@nestjs/swagger'
+
+import { UseAuthGuard } from '../auth/misc/jwt.decorator'
+import { User } from '../auth/misc/user.decorator'
+import { ResumeService } from './resume.service'
+
+@ApiTags('resume')
+@Controller('resume')
```
Comment (palmcm) [inline apps/api/src/modules/resume/resume.controller.ts:17]: please add s

File: apps/api/src/modules/resume/resume.controller.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+export class ResumeController {
+  constructor(private readonly resumeService: ResumeService) {}
+
+  @Post()
+  @UseAuthGuard()
+  @UseAuthGuard(UserType.ACTOR)
```
Comment (palmcm) [inline apps/api/src/modules/resume/resume.controller.ts:23]:
```suggestion
  @UseAuthGuard(UserType.ACTOR)
```

File: apps/api/src/modules/resume/resume.respository.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+export class ResumeRepository {
+  constructor(private prisma: PrismaService) {}
+
+  async createResume(postResumeDto: PostResumeDto, user: JwtDto) {
+    const actor = await this.prisma.actor.findUnique({
+      where: { actorId: user.userId },
+    })
+    if (!actor) {
+      throw new NotFoundException('Actor not found')
+    }
```
Comment (palmcm) [inline apps/api/src/modules/resume/resume.respository.ts:15]: @miello do we need to do this

File: apps/api/src/modules/resume/resume.service.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import { ResumeRepository } from './resume.respository'
+
+@Injectable()
+export class ResumeService {
+  constructor(private readonly repository: ResumeRepository) {}
+
+  private validateResumeDto() {
+    return true
+  }
```
Comment (palmcm) [inline apps/api/src/modules/resume/resume.service.ts:12]: maybe remove this

File: apps/api/src/modules/resume/resume.controller.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  @ApiUnauthorizedResponse({ description: 'User is not logged in' })
```
Comment (palmcm) [inline apps/api/src/modules/resume/resume.controller.ts:24]: Maybe add `@ApiCreatedResponse({ type: someDto })` for docs

**Review by palmcm — CHANGES_REQUESTED** [review]

**Review by palmcm — APPROVED** [review]

File: apps/api/src/modules/resume/resume.respository.ts (hunk 3 above)
Comment (miello) [reply apps/api/src/modules/resume/resume.respository.ts:15]: Can we do checking on ResumeService instead ?

**Review by miello — APPROVED** [review]

**Pull request merged**